### PR TITLE
Make sure no nil lists are passed to TF

### PR DESCRIPTION
**Pull request by VenelinMartinov**
Addresses https://github.com/pulumi/pulumi-aws/issues/3427

We should make sure we send the upstream provider an empty array where MaxItemsOne properties are expected instead of a nil, since that is what TF does.

### CODE REVIEW

File: pkg/tfbridge/schema.go [hunk 1]
```diff
@@ -655,6 +655,15 @@ func (ctx *conversionContext) makeObjectTerraformInputs(
 		return nil, err
 	}
 
+	// Iterate over the TF schema and add an empty list for each list type not in the results.
+	tfs.Range(func(key string, value shim.Schema) bool {
+		schema := tfs.Get(key)
+		if schema.Type() == shim.TypeList && result[key] == nil {
```
Comment (iwahbe) [inline pkg/tfbridge/schema.go:661]: Does terraform distinguish between `nil` and missing?

**Comment by VenelinMartinov** [comment]
After a discussion with @t0yv0 I dug a bit into the code related to `makeTerraformInput` trying to trace why we are not converting the missing input into an empty array for TF.

He suggested this is likely related to `MaxItemsOne` since we don't have a concept of `empty array` for these properties, while TF does.

`Create` calls `UnmarshalTerraformConfig`: https://github.com/pulumi/pulumi-terraform-bridge/blob/096668ca994bd09ad7b55b889571bc2945fbe695/pkg/tfbridge/provider.go#L1000

which one call down calls `makeTerraformInputs`: https://github.com/pulumi/pulumi-terraform-bridge/blob/e87770530db612b4736e8b640677e928751875bd/pkg/tfbridge/schema.go#L1233

Which then calls `makeObjectTerraformInputs`: https://github.com/pulumi/pulumi-terraform-bridge/blob/e87770530db612b4736e8b640677e928751875bd/pkg/tfbridge/schema.go#L546

which only iterates over the `news`: https://github.com/pulumi/pulumi-terraform-bridge/blob/e87770530db612b4736e8b640677e928751875bd/pkg/tfbridge/schema.go#L605

At this point the `forward` property is not in the `news` since it was not specified in the pulumi program.

This means that the code which is meant to fill the empty array in place of the `MaxItemsOne` nil is never called:
https://github.com/pulumi/pulumi-terraform-bridge/blob/e87770530db612b4736e8b640677e928751875bd/pkg/tfbridge/schema.go#L357-L362

**Comment by t0yv0** [comment]
2 sounds exact to me. The transformation is schema-aware and should be walking object fields even if they're not present in the source PropertyMap but are present in the schema, so it can apply the unflattening treatment as necessary.

File: pkg/tfbridge/schema.go (hunk 1 above)
Comment (VenelinMartinov) [reply pkg/tfbridge/schema.go]: Great question but I doubt it given the way they use them: https://github.com/hashicorp/terraform-provider-aws/blob/3645bd03060df19f5cf29c132e26f7e40b56230f/internal/service/elbv2/listener.go#L425C1-L427C3

**Comment by VenelinMartinov** [comment]
I still need some tests for this.

**Review by iwahbe — APPROVED** [review]
Looks good to me. LGTM

File: pkg/tfbridge/schema.go (hunk 1 above)
Comment (t0yv0) [reply pkg/tfbridge/schema.go]: In maps yes, in objects no, an object having an optional field F will have it nil if it's missing, there's no other way for it to have it missing.

**Pull request merged**